File: image-service/src/main/java/com/chickpic/microservices/image/model/Image.java
package com.chickpic.microservices.image.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "image")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Getter
@Setter
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String title;
    private String description;
    private String country;
    private String city;
    private double lat;
    private double lng;
    private String fileName;
}

File: src/app/model/image.ts
export interface ImageInfo {
    title: string;
    description: string;
    country: string;
    city: string;
    lat: number;
    lng: number;
    fileName?: string;
    fileUrl?: string;
}

export type Image = {
    title: string;
    description: string;
    fileName: string;
    fileUrl: string;
};

export type ImageByLocation = {
    country: string;
    cities: {
        city: string;
        images: Image[];
        lat: number;
        lng: number;
    }[];
};
